octree/basic.go

```diff
@@ -10,14 +10,18 @@ import (
 	pc "go.viam.com/rdk/pointcloud"
 )
 
+const (
+	maxRecursionDepth = 1000
```

Comment:
Please add a comment justifying this recursion depth (e.g. that it was tested on a pi and lets us support high granularity of points)